author: bearinmindcat <bearincrypto@gmail.com>

--- a/chrome/browser/sync/sync_service_factory.cc
+++ b/chrome/browser/sync/sync_service_factory.cc
@@ -300,6 +300,8 @@
 
   builder.SetDataTypeStoreService(
       DataTypeStoreServiceFactory::GetForProfile(profile));
+  builder.SetSecurityEventRecorder(
+      SecurityEventRecorderFactory::GetForProfile(profile));
 
 #if BUILDFLAG(ENABLE_EXTENSIONS_CORE)
   builder.SetExtensionSyncService(ExtensionSyncService::Get(profile));
@@ -396,7 +398,114 @@
     std::optional<syncer::CreateHttpPostProviderFactory>
         create_http_post_provider_factory_for_test,
     content::BrowserContext* context) {
+  if (create_http_post_provider_factory_for_test.has_value()) {
+    CHECK_IS_TEST();
+  }
+
+  syncer::SyncServiceImpl::InitParams init_params;
+
+  Profile* profile = Profile::FromBrowserContext(context);
+  CHECK(profile);
+
+  // Incognito, guest, or system profiles aren't relevant for Sync, and
+  // no SyncService should be created for those types of profiles.
+  CHECK(profiles::IsRegularUserProfile(profile));
+
+  init_params.sync_client = BuildSyncClient(profile);
+  init_params.url_loader_factory = profile->GetDefaultStoragePartition()
+                                       ->GetURLLoaderFactoryForBrowserProcess();
+  init_params.create_http_post_provider_factory =
+      std::move(create_http_post_provider_factory_for_test)
+          .value_or(base::BindRepeating(
+              [](const std::string& user_agent,
+                 std::unique_ptr<network::PendingSharedURLLoaderFactory>
+                     pending_url_loader_factory)
+                  -> std::unique_ptr<syncer::HttpPostProviderFactory> {
+                return std::make_unique<syncer::HttpBridgeFactory>(
+                    user_agent, std::move(pending_url_loader_factory));
+              }));
+  init_params.network_connection_tracker =
+      content::GetNetworkConnectionTracker();
+  init_params.channel = chrome::GetChannel();
+  init_params.debug_identifier = profile->GetDebugName();
+  init_params.os_crypt_async = g_browser_process->os_crypt_async();
+
+  bool local_sync_backend_enabled = false;
+  // Only check the local sync backend pref on the supported platforms of
+  // Windows, Mac and Linux.
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+  syncer::SyncPrefs prefs(profile->GetPrefs());
+  local_sync_backend_enabled = prefs.IsLocalSyncEnabled();
+  base::UmaHistogramBoolean("Sync.Local.Enabled2", local_sync_backend_enabled);
+
+  if (local_sync_backend_enabled) {
+    base::FilePath local_sync_backend_folder =
+        init_params.sync_client->GetLocalSyncBackendFolder();
+
+    // If the user has not specified a folder and we can't get the default
+    // roaming profile location the sync service will not be created.
+    base::UmaHistogramBoolean("Sync.Local.RoamingProfileUnavailable2",
+                              local_sync_backend_folder.empty());
+    if (local_sync_backend_folder.empty()) {
       return nullptr;
+    }
+  }
+#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+
+  if (!local_sync_backend_enabled) {
+    // Always create the GCMProfileService instance such that we can listen to
+    // the profile notifications and purge the GCM store when the profile is
+    // being signed out.
+    gcm::GCMProfileServiceFactory::GetForProfile(profile);
+
+    AboutSigninInternalsFactory::GetForProfile(profile);
+  }
+
+  auto sync_service =
+      std::make_unique<syncer::SyncServiceImpl>(std::move(init_params));
+  sync_service->Initialize(CreateControllers(profile, sync_service.get()));
+
+  // Notify the PasswordStore of complete initialisation to resolve a circular
+  // dependency.
+  auto profile_password_store = ProfilePasswordStoreFactory::GetForProfile(
+      profile, ServiceAccessType::EXPLICIT_ACCESS);
+  // PasswordStoreInterface may be null in tests.
+  if (profile_password_store) {
+    profile_password_store->OnSyncServiceInitialized(sync_service.get());
+  }
+
+  auto account_password_store = AccountPasswordStoreFactory::GetForProfile(
+      profile, ServiceAccessType::EXPLICIT_ACCESS);
+
+  if (account_password_store) {
+    account_password_store->OnSyncServiceInitialized(sync_service.get());
+  }
+
+  // Notify PasswordReceiverService of complete initialization to resolve a
+  // circular dependency.
+  password_manager::PasswordReceiverService* password_receiver_service =
+      PasswordReceiverServiceFactory::GetForProfile(profile);
+  if (password_receiver_service) {
+    password_receiver_service->OnSyncServiceInitialized(sync_service.get());
+  }
+
+  SendTabToSelfSyncServiceFactory::GetForProfile(profile)
+      ->OnSyncServiceInitialized(sync_service.get());
+
+  collaboration::CollaborationServiceFactory::GetForProfile(profile)
+      ->OnSyncServiceInitialized(sync_service.get());
+
+  // Allow sync_preferences/ components to use SyncService.
+  sync_preferences::PrefServiceSyncable* pref_service =
+      PrefServiceSyncableFromProfile(profile);
+  pref_service->OnSyncServiceInitialized(sync_service.get());
+
+  if (GoogleGroupsManager* groups_updater_service =
+          GoogleGroupsManagerFactory::GetForBrowserContext(profile)) {
+    groups_updater_service->OnSyncServiceInitialized(sync_service.get());
+  }
+
+  return sync_service;
 }
 
 }  // namespace
@@ -467,6 +576,7 @@
   DependsOn(PasswordSenderServiceFactory::GetInstance());
   DependsOn(ProfilePasswordStoreFactory::GetInstance());
 
+  DependsOn(SecurityEventRecorderFactory::GetInstance());
   DependsOn(SendTabToSelfSyncServiceFactory::GetInstance());
   DependsOn(SharingMessageBridgeFactory::GetInstance());
 #if !BUILDFLAG(IS_ANDROID)
