diff --git a/chrome/browser/download/download_crx_util.cc b/chrome/browser/download/download_crx_util.cc
--- a/chrome/browser/download/download_crx_util.cc
+++ b/chrome/browser/download/download_crx_util.cc
@@ -27,6 +27,11 @@
 #include "extensions/common/extension_urls.h"
 #include "extensions/common/user_script.h"
 
+#if BUILDFLAG(IS_ANDROID)
+#include "chrome/browser/ui/android/tab_model/tab_model.h"
+#include "chrome/browser/ui/android/tab_model/tab_model_list.h"
+#endif
+
 #if !BUILDFLAG(IS_ANDROID)
 #include "chrome/browser/ui/browser_window/public/browser_window_interface.h"
 #include "chrome/browser/ui/browser_window/public/create_browser_window.h"
@@ -64,6 +69,16 @@
   content::WebContents* web_contents =
       content::DownloadItemUtils::GetWebContents(
           const_cast<DownloadItem*>(&download_item));
+#if BUILDFLAG(IS_ANDROID)
+  if (!web_contents) {
+    for (TabModel* model : TabModelList::models()) {
+      if (model->IsActiveModel()) {
+        web_contents = model->GetActiveWebContents();
+        break;
+      }
+    }
+  }
+#endif
   if (!web_contents) {
     BrowserWindowInterface* browser =
         extensions::browser_window_util::GetLastActiveBrowserWithProfile(
