diff --git a/chrome/browser/extensions/api/BUILD.gn b/chrome/browser/extensions/api/BUILD.gn
--- a/chrome/browser/extensions/api/BUILD.gn
+++ b/chrome/browser/extensions/api/BUILD.gn
@@ -53,6 +53,7 @@
     "//chrome/browser/extensions/api/reading_list",
     "//chrome/browser/extensions/api/runtime",
     "//chrome/browser/extensions/api/search",
+    "//chrome/browser/extensions/api/settings_overrides",
     "//chrome/browser/extensions/api/tab_capture",
     "//chrome/browser/extensions/api/tab_groups",
     "//chrome/browser/extensions/api/web_navigation",
@@ -77,7 +78,6 @@
       "//chrome/browser/extensions/api/passwords_private",
       "//chrome/browser/extensions/api/resources_private",
       "//chrome/browser/extensions/api/safe_browsing_private",
-      "//chrome/browser/extensions/api/settings_overrides",
       "//chrome/browser/extensions/api/side_panel",
       "//chrome/browser/extensions/api/webrtc_desktop_capture_private",
     ]
diff --git a/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc b/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc
--- a/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc
+++ b/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc
@@ -29,6 +29,7 @@
 
 // The following are not supported in the experimental desktop-android build.
 // TODO(https://crbug.com/356905053): Enable these APIs on desktop-android.
+#include "chrome/browser/extensions/api/settings_overrides/settings_overrides_api.h"
 #if BUILDFLAG(ENABLE_EXTENSIONS)
 #include "build/build_config.h"
 #include "build/chromeos_buildflags.h"
@@ -42,7 +43,6 @@
 #include "chrome/browser/extensions/api/networking_private/networking_private_ui_delegate_factory_impl.h"
 #include "chrome/browser/extensions/api/passwords_private/passwords_private_delegate_factory.h"
 #include "chrome/browser/extensions/api/passwords_private/passwords_private_event_router_factory.h"
-#include "chrome/browser/extensions/api/settings_overrides/settings_overrides_api.h"
 #include "chrome/browser/extensions/api/settings_private/settings_private_event_router_factory.h"
 #include "chrome/browser/extensions/api/side_panel/side_panel_service.h"
 #include "components/safe_browsing/buildflags.h"
@@ -106,6 +106,7 @@
   extensions::TabsWindowsAPI::GetFactoryInstance();
   extensions::WebNavigationAPI::GetFactoryInstance();
 
+  extensions::SettingsOverridesAPI::GetFactoryInstance();
 #if BUILDFLAG(ENABLE_EXTENSIONS)
   extensions::ActivityLogAPI::GetFactoryInstance();
   extensions::AutofillPrivateEventRouterFactory::GetInstance();
@@ -138,7 +139,6 @@
   extensions::SafeBrowsingPrivateEventRouterFactory::GetInstance();
 #endif
   extensions::SettingsPrivateEventRouterFactory::GetInstance();
-  extensions::SettingsOverridesAPI::GetFactoryInstance();
   extensions::SidePanelService::GetFactoryInstance();
 #if BUILDFLAG(IS_CHROMEOS)
   extensions::TerminalPrivateAPI::GetFactoryInstance();
diff --git a/chrome/browser/extensions/api/settings_overrides/BUILD.gn b/chrome/browser/extensions/api/settings_overrides/BUILD.gn
--- a/chrome/browser/extensions/api/settings_overrides/BUILD.gn
+++ b/chrome/browser/extensions/api/settings_overrides/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//extensions/buildflags/buildflags.gni")
 
-assert(enable_extensions)
+assert(enable_extensions_core)
 
 source_set("settings_overrides") {
   sources = [
diff --git a/chrome/common/extensions/api/_manifest_features.json b/chrome/common/extensions/api/_manifest_features.json
--- a/chrome/common/extensions/api/_manifest_features.json
+++ b/chrome/common/extensions/api/_manifest_features.json
@@ -17,8 +17,7 @@
   },
   "chrome_settings_overrides": {
     "channel": "stable",
-    "extension_types": ["extension"],
-    "platforms": ["win", "mac"]
+    "extension_types": ["extension"]
   },
   "file_browser_handlers": {
     "channel": "stable",
